src/app/MessageDef.h

```diff
@@ -0,0 +1,1196 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2016-2017 Nest Labs, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+/**
+ *    @file
+ *      This file defines CHIP interaction model message parsers and encoders
+ *      Interaction model profile.
+ *
+ */
+
+#ifndef _CHIP_INTERACTION_MODEL_MESSAGE_DEF_H
+#define _CHIP_INTERACTION_MODEL_MESSAGE_DEF_H
+
+#include <core/CHIPCore.h>
+#include <core/CHIPTLV.h>
+#include <support/CodeUtils.h>
+#include <support/DLLUtil.h>
+#include <support/logging/CHIPLogging.h>
+#include <util/basic-types.h>
+
+namespace chip {
+namespace app {
+/**
+ *  @brief
+ *    The CHIP interaction model message types.
+ *
+ *  These values are called out in CHIP Interaction Model: Encoding Specification
+ *
+ */
+enum
+{
+    kMsgType_SubscribeRequest      = 0x01,
+    kMsgType_ReadRequest           = 0x02,
+    kMsgType_ReportData            = 0x03,
+    kMsgType_WriteRequest          = 0x04,
+    kMsgType_WriteResponse         = 0x05,
+    kMsgType_InvokeCommand  = 0x06,
+    kMsgType_InvokeCommandResponse = 0x07,
+};
+
+static inline bool IsVersionNewer(const chip::DataVersion & aVersion, const chip::DataVersion & aReference)
+{
+    return (aVersion != aReference);
+}
+
+static inline bool IsVersionNewerOrEqual(const chip::DataVersion & aVersion, const chip::DataVersion & aReference)
```

Comment:
Drop this method - it does not do any comparison.

Reply:
removed, thanks